common/src/View/MapDocumentCommandFacade.cpp

```diff
@@ -871,7 +871,7 @@ namespace TrenchBroom {
         void MapDocumentCommandFacade::performSetTextureCollections(const IO::Path::List& paths) {
             const Model::NodeList nodes(1, m_world);
             Notifier1<const Model::NodeList&>::NotifyBeforeAndAfter notifyNodes(nodesWillChangeNotifier, nodesDidChangeNotifier, nodes);
-            Notifier0::NotifyAfter notifyTextureCollections(textureCollectionsDidChangeNotifier);
+            Notifier0::NotifyBeforeAndAfter notifyTextureCollections(textureCollectionsWillChangeNotifier, textureCollectionsDidChangeNotifier);
             
             m_game->updateTextureCollections(m_world, paths);
             unsetTextures();
```

Comment:
In the next line, the notifiers will be called again, which may result in some inconsistencies. That's why I try to send all notifications from this class as opposed to `MapDocument` if at all possible.